packages/core/admin/index.js

```diff
@@ -70,15 +41,23 @@ async function build({ plugins, dir, env, options, optimize, forceBuild }) {
 
   const pluginsPath = Object.keys(plugins).map(pluginName => plugins[pluginName].pathToPlugin);
 
-  const config = getCustomWebpackConfig(dir, {
-    entry,
-    pluginsPath,
+  // Either use the tsconfig file from the generated app or the one inside the .cache folder
+  // so we can develop plugins in TS while being in a JS app
+  const tsConfigFilePath = useTypeScript
+    ? path.join(appDir, 'src', 'admin', 'tsconfig.json')
```

Comment:
Shoudln't we have only one tsconfig or a tsconfig.admin.json and tsconfig.server.json at the root of the projects ? That would be simpler to use for devs IMO 🤔

Reply:
Linking this discussion: https://github.com/strapi/strapi/issues/12927